.github/workflows/pre-commit.yml

```diff
@@ -31,30 +32,22 @@ jobs:
           sudo apt-get update -qq || (sudo rm /etc/apt/sources.list.d/* && sudo apt-get update -yqq)
           sudo apt-get install -y libcurl4-openssl-dev
 
-      - name: Set up Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.8"
-          architecture: "x64"
-
       - name: Set up environment
         run: |
-          echo "UBUNTU_VERSION=`grep DISTRIB_RELEASE /etc/lsb-release | sed 's/.*=//g'`" >> $GITHUB_ENV
-          mkdir -p .local/R/site-packages
-          echo "R_LIBS_USER=`pwd`/.local/R/site-packages" >> $GITHUB_ENV
+          echo "LINUX_VERSION=$(uname -a)" >> $GITHUB_ENV
+          echo "R_VERSION=$(R --version | head -n 1)" >> $GITHUB_ENV
 
       - name: Cache R packages
         uses: actions/cache@v2
         if: startsWith(runner.os, 'Linux')
         with:
-          path: ${{env.R_LIBS_USER}}
-          key: precommit-${{env.UBUNTU_VERSION}}-renv-${{ hashFiles('.pre-commit.r_requirements.txt') }}-
+          path: /usr/local/lib/R/site-library
+          key: precommit-${{env.LINUX_VERSION}}-${{env.R_VERSION}}-renv-${{ hashFiles('.pre-commit.r_requirements.txt') }}-
           restore-keys: |
-            precommit-${{env.UBUNTU_VERSION}}-renv-
+            precommit-${{env.LINUX_VERSION}}-${{env.R_VERSION}}-renv-
 
       - name: Install R packages
         run: |
-          if (!requireNamespace("renv", quietly = TRUE)) install.packages("renv")
           renv::restore()
           renv::install(scan(".pre-commit.r_requirements.txt", what=character(), sep="\n"))
```

Comment:
do we need to install r_requirements here again? This is done in the pulled docker image already, no?

Reply:
Once we merge #345 then we can use that image, yes -- there is an open question as to how we should deal with updating requirements, I think I lean towards attempting to install again in case the requirements file has changed.

Reply:
If the dockerfile for the pulled image is there, wouldn't it be easier to just add to that manually if something needs changing? Especially R packages usually take some time to install. That approach is not error proof though and may be more difficult to maintain... can we check what's already been installed here instead of installing everything?

Reply:
Just added a script that checks what's been installed and only installs that.

Reply:
awesome!